common/src/main/java/io/pravega/common/util/CollectionHelpers.java

```diff
@@ -61,6 +62,55 @@
         return -1;
     }
 
+    /**
+     * Performs a binary search on the given IndexedMap.
+     *
+     * @param source      The IndexedMap to search on. The Entries in this map must be sorted by Key, otherwise the outcome
+     *                    of this method will be undefined.
+     * @param keys        A Collection containing Keys to search on. This collection need not be sorted.
+     * @param result      A Map that will contain the results. Only Keys that are present in the given source will be populated
```

Comment:
This could be a little more generic if it returned a collection of indexes. That would separate it from ValueType, and mean `IndexedMap` could just be `Indexed`. The signature would look like:
```java
public static <T extends Comparable<T>> List<Integer> binarySearch(Indexed<T> source, Collection<T> locate)
```
Where indexed looks something like: 
```java
public interface Indexed<Type> {
    Type get(int position);
}
```

Reply:
I agree with you that it could be a bit more generic. But that won't make it any more useful for me. The reason I pass a Map as a parameter is because I collect results in it from various calls. If I have it return a List, then I'd have to re-iterate through that list and fetch the values. That's one extra loop that I prefer not to add to my code.